Corona-Warn-App/src/main/java/de/rki/coronawarnapp/task/InternalTaskState.kt

```diff
@@ -0,0 +1,31 @@
+package de.rki.coronawarnapp.task
+
+import de.rki.coronawarnapp.task.TaskState.ExecutionState
+import kotlinx.coroutines.Deferred
+import org.joda.time.Instant
+import java.util.UUID
+import kotlin.reflect.KClass
+
+internal data class InternalTaskState(
+    internal val id: UUID = UUID.randomUUID(),
+    override val request: TaskRequest,
+    override val createdAt: Instant,
+    override val startedAt: Instant? = null,
+    override val completedAt: Instant? = null,
+    override val error: Throwable? = null,
+    override val result: Task.Result? = null,
+    internal val config: TaskFactory.Config,
+    internal val deferred: Deferred<Task.Result>,
```

Comment:
result vs deferred? What's the difference as it is of the same result type. Maybe just a naming "issue"

Reply:
`Deferred value is a non-blocking cancellable future &mdash; it is a [Job] with a result.`

Renaming to `job`.